### PR TITLE
Bug in loc raising KeyError when MultiIndex columns has only one level

### LINKED ISSUE #29749: Multiindex with single level no longer allows selecting columns
This works fine in pandas 0.22.0 but fails in 0.25.3 It appears a bug may have been introduced.

I have some old code that accidentally created a MultiIndex with a single level instead of a regular index, but everything worked fine with pandas 0.22.0. However, in new versions of pandas, that MultiIndex breaks pandas ability to select columns and gives an extremely misleading error message. In this case, I don't need the MultiIndex, so I can just remove that; but I think this code ought to work. (And it did work in the old 0.22.0 version).  

```
import numpy as np
import pandas as pd

names = ['FirstColumn', 'SecondColumn']
data = np.array([[5,6],[7,8]])
df = pd.DataFrame(data, columns = [names]) #Bug: the brackets around "[names]" creates a
#multi-index but that was unintentional. (It's still legal, though). 
#But "df.head()" and "df.describe()" both look normal so you can't see anything is wrong. 

#All of the following work as expected in 0.22.0 but give misleading errors in 0.25.3.
#These all work with a regular index, but the 0.25.3 MultIndex doesn't like it. 
df['FirstColumn'] #ERROR! 
df.FirstColumn #ERROR! 
df.loc[:,'FirstColumn'] #ERROR! 

#This works, but it shouldn't be necessary to do this. The above syntax is more standard.
df.xs('FirstColumn', axis=1, level=0)
```
Those failing statements give misleading errors about ```only integer scalar arrays can be converted to a scalar index```  but that code looks like it ought to work, and it did work fine in the previous versions of pandas. Can we make the standard methods of selecting columns work with the MultiIndex?

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #29749
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

@jbrockmendel I think we should check for MultiIndex instead of level number here

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 1]
```diff
@@ -659,3 +659,15 @@ def test_getitem_non_found_tuple():
     )
     with pytest.raises(KeyError, match=r"\(2\.0, 2\.0, 3\.0\)"):
         df.loc[(2.0, 2.0, 3.0)]
+
+
+def test_loc_multiindex_columns_one_level():
```
Comment (jbrockmendel) [inline pandas/tests/indexing/multiindex/test_loc.py:664]: deep down this is a `DataFrame.__getitem__` test

**Comment by jbrockmendel** [comment]
comment on test placement, otherwise LGTM

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexing/multiindex/test_loc.py]: Thx, moved. Thought about were to put it, but looking at the place were the modification was necessary should have answered it easily

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -3151,8 +3151,8 @@ def get_indexer(
                 )
 
             indexer = self._engine.get_indexer(target._get_engine_target())
-
-        return ensure_platform_int(indexer)
+        x = ensure_platform_int(indexer)
```
Comment (ApamNapat) [inline pandas/core/indexes/base.py:3154]: This variable adds nothing but noise to the code

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -4883,8 +4883,8 @@ def get_indexer_non_unique(self, target):
             tgt_values = np.asarray(target)
         else:
             tgt_values = target._get_engine_target()
-
-        indexer, missing = self._engine.get_indexer_non_unique(tgt_values)
+        engine = self._engine
+        indexer, missing = engine.get_indexer_non_unique(tgt_values)
```
Comment (ApamNapat) [inline pandas/core/indexes/base.py:4887]: Same as above, this line was perfectly fine

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (phofl) [reply pandas/core/indexes/base.py]: Yep this was accidential

File: pandas/core/indexes/base.py (hunk 3 above)
Comment (phofl) [reply pandas/core/indexes/base.py]: Same here

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl very nice
keep em coming!

**Pull request merged**